### DISCUSSION TITLE
struggling with dialogue_ended?

### DISCUSSION THREAD

**Discussion by ElleanoreB** [nathanhoad/godot_dialogue_manager#351] [Q&A] [answered]
Hi, 
Newish to godot and coding is not my strongest suit, but was pleased to be able to get dialogue manager up and running and even figure my own custom dialogue box designs. 
However, I'm struggling with being able to tell when dialogue has ended. I've looked it up and everywhere I'm being told to use the dialogue_ended signal, but I don't seem to be able to actually get that signal when the dialogue ends. 
It's possible that I'm just being dumb, I've checked the tutorials for dialogue manager on youtube and none of them actually show dialogue_ended being used in action so I'm really confused as to what I'm supposed to be doing to get it to work.

Thanks for any help, excited to continue using Dialogue Manager in my project once I get over this bump!

**Comment by Selenyhr** [comment, marked as answer]
Hiya!

Since Dialogue Manager automatically registers itself as an Autoload, the way you usually do some sort of processing on the `dialogue_ended` signal is the following:

```gdscript
func _ready() -> void:
	DialogueManager.dialogue_ended.connect(_on_dialogue_manager_dialogue_ended)


func _on_dialogue_manager_dialogue_ended(resource: DialogueResource) -> void:
	# Do things here
```

This is the standard way of connecting signals in Godot. As you can see in this case, there is a `DialogueResource` argument that is being passed. This corresponds to the file that has ended.

This signal is emitted once Dialogue Manager encounters either the implicit end of a title section (so a new title, or the end of the file), or a `=> END` instruction.

**Reply by ElleanoreB** [reply]
Ah! 
Thank you so much for dispelling my confusion, I was able to get it working this way.
I'll mark this as answered 👍

**Reply by ghost** [reply]
holy ive been looking evrywhere for an answwer thank you
